src/ImageDeleteModal.jsx

```diff
@@ -17,115 +17,98 @@ function sortTags(a, b) {
     return a.localeCompare(b);
 }
 
-export class ImageDeleteModal extends React.Component {
-    static contextType = DialogsContext;
-
-    constructor(props) {
-        super(props);
-
-        const tags = {};
-        const repoTags = this.props.imageWillDelete.RepoTags ? this.props.imageWillDelete.RepoTags : [];
-        repoTags.sort(sortTags).forEach((x, i) => {
-            tags[x] = (i === 0);
-        });
-
-        this.state = {
-            tags: tags,
-        };
-
-        this.onValueChanged = this.onValueChanged.bind(this);
-        this.pickAll = this.pickAll.bind(this);
-    }
-
-    onValueChanged(item, value) {
-        this.setState(prev => {
-            const tags = prev.tags;
-            tags[item] = value;
-            return { tags: tags };
-        });
-    }
-
-    pickAll() {
-        this.setState(prev => {
-            const tags = prev.tags;
-            Object.keys(tags).forEach(item => { tags[item] = true });
-            return { tags: tags };
-        });
-    }
-
-    handleRemoveImage(tags, all) {
-        const Dialogs = this.context;
-        const image = this.props.imageWillDelete;
-
+const ImageDeleteModal = ({ imageWillDelete, onAddNotification }) => {
+    const Dialogs = useDialogs();
+    const repoTags = imageWillDelete.RepoTags ? imageWillDelete.RepoTags : [];
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20221130-134924-205df97d-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)